src/transformers/pipelines.py

```diff
@@ -1435,17 +1457,31 @@ def group_sub_entities(self, entities: List[dict]) -> dict:
             entities (:obj:`dict`): The entities predicted by the pipeline.
         """
         # Get the first entity in the entity group
-        entity = entities[0]["entity"]
-        scores = np.mean([entity["score"] for entity in entities])
+        entity = entities[0]["entity"].split("-")[-1]
+        scores = np.nanmean([entity["score"] for entity in entities])
         tokens = [entity["word"] for entity in entities]
 
         entity_group = {
             "entity_group": entity,
             "score": np.mean(scores),
-            "word": self.tokenizer.convert_tokens_to_string(tokens),
+            "word": self.convert_tokens_to_string(tokens),
         }
         return entity_group
 
+    def is_subword_fn(self, token: str) -> bool:
+        if token.startswith("##"):
+            return True
+        return False
+
+    def convert_tokens_to_string(self, tokens):
+        """ Converts a sequence of tokens (string) in a single string. """
+        if hasattr(self.tokenizer, "convert_tokens_to_string"):
+            # fast tokenizers dont have convert_tokens_to_string?!
+            return self.tokenizer.convert_tokens_to_string(tokens)
+        else:
+            out_string = " ".join(tokens).replace(" ##", "").strip()
+            return out_string
```

Comment:
This too is very specific to BERT

Reply:
same here. can be overridden by tokenizer.convert_tokens_to_string. I recently saw the [BatchEncoding.token_to_word](https://huggingface.co/transformers/main_classes/tokenizer.html#transformers.BatchEncoding.token_to_word) maybe I can use that, will take a look

Reply:
Can we use the `token_to_word` if it's available, `convert_tokens_to_string` otherwise?